jgrapht-core/src/main/java/org/jgrapht/alg/tour/HeldKarpTSP.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * (C) Copyright 2017-2017, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.tour;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.interfaces.TSPAlgorithm;
+import org.jgrapht.graph.GraphWalk;
+
+import java.util.*;
+
+/**
+ * A dynamic programming algorithm for the TSP problem.
```

Comment:
Do you have a reference as to which paper/article you used to create this particular implementation (please include)? Or did you solely use the wiki description?

Reply:
I only used the wiki description.